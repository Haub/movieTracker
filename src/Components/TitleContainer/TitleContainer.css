.card-container {
  display: flex;
  flex-wrap: wrap;
  margin: 2%
}