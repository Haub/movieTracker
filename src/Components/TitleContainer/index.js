import React from 'react';
import TitleCard from '../TitleCard';
import './TitleContainer.css'

const TitleContainer = ({ movies, name, search }) => {
  let uuidv4 = require("uuid/v4")
  const favoriteCount = movies.filter(movie => movie.favorite)

  const filteredMovies = search && search.length 
    ? movies.filter(movie => movie.title.includes(search))
    : movies
  
  const displayMovies = filteredMovies.map(movie => (
    <TitleCard 
      {...movie}
      key={uuidv4()}
    />
  ))
  const searchActive = search && search.length > 0 ? true : false

  
  if (favoriteCount.length >= 4 && name === 'Recent Favorites') {
    return(
      <div className={`title-container ${searchActive ? 'pad' : ''}`}>
        <div className='container-name'>
          <div className='favorites-page'>
            <h3 className='title fade'>{name}</h3>
            <h3 className='title'>(see all)</h3>
          </div>
        </div>
        <div className='card-container-favorites'>
          {displayMovies}
        </div>
      </div>
    )
  } else {
    return(
      <div className={`title-container ${searchActive ? 'pad' : ''}`}>
        <div className='container-name'>
        {
          favoriteCount.length &&
          <div>
            <h3 className='title fade'>{name}</h3>
          </div>
        }
        </div>
        <div className='card-container'>
          {displayMovies}
        </div>
      </div>
    )
  }
}

export default TitleContainer;


