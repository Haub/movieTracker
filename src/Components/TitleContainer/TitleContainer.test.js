import React from 'react';
import TitleContainer from '../TitleContainer';
import { shallow } from 'enzyme';

describe('TITLE CONTAINER', () => {

  it('should match the snapshot', () => {
    const mockMovies = [{title: 'Venom'}, {title: 'Halloween'}]
    const mockName = 'Venom';
    const mockSearch = 'Ve';
    const wrapper = shallow(
      <TitleContainer
        movies={mockMovies}
        name={mockName}
        search={mockSearch}
      />
    )
    expect(wrapper).toMatchSnapshot();
  })
})