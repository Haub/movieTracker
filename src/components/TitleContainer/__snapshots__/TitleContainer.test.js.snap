// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TITLE CONTAINER should match the snapshot 1`] = `
<div
  className="pad"
>
  <div
    className="container-name"
  />
  <div
    className="card-container"
  >
    <Connect(TitleCard)
      key="eef7a903-1726-48c7-860b-feb25587089d"
      title="Venom"
    />
  </div>
</div>
`;
